source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -15,17 +16,37 @@ namespace HttpFilters {
 namespace Lua {
 
 namespace {
+const std::string DEPRECATED_LUA_NAME = "envoy.lua";
+
 const ProtobufWkt::Struct& getMetadata(Http::StreamFilterCallbacks* callbacks) {
   if (callbacks->route() == nullptr || callbacks->route()->routeEntry() == nullptr) {
     return ProtobufWkt::Struct::default_instance();
   }
   const auto& metadata = callbacks->route()->routeEntry()->metadata();
-  const auto& filter_it = metadata.filter_metadata().find(HttpFilterNames::get().Lua);
-  if (filter_it == metadata.filter_metadata().end()) {
-    return ProtobufWkt::Struct::default_instance();
+
+  {
+    const auto& filter_it = metadata.filter_metadata().find(HttpFilterNames::get().Lua);
+    if (filter_it != metadata.filter_metadata().end()) {
+      return filter_it->second;
+    }
   }
-  return filter_it->second;
+
+  // TODO(zuercher): Remove this block when deprecated filter names are removed.
+  {
+    const auto& filter_it = metadata.filter_metadata().find(DEPRECATED_LUA_NAME);
+    if (filter_it != metadata.filter_metadata().end()) {
+      // Use the non-throwing check here because this happens at request time.
```

Comment:
For someone that upgrades without fixing this and uses this in the hot path, this will blow up the logs I think? I think we probably need to do some type of warn once? I'm not sure the easiest/cheapest way of handling this. :(

Reply:
Right. I think I have an idea for this.

Reply:
Latest version will increment the stat every time but only logs once per process.